rep-2008.rst

```diff
@@ -0,0 +1,387 @@
+
+
+.. list-table:: 
+   :widths: 25 25 50
+   :header-rows: 1
+
+   * - ``ament`` ROS 2 build system extension
+     - variable
+     - description
+   * - `ament_hardware_acceleration <https://github.com/ros-acceleration/ament_hardware_acceleration/>`_
+     - ``ROS_ACCELERATION``
+     - This CMake variable will evaluate to ``True`` when targeting *any* of the supported ROS 2-enabled accelerators (see ``mixins`` enablement below). Use while integrating acceleration kernels in a technology and vendor-agnostic manner.
+   * - `ament_vitis <https://github.com/ros-acceleration/ament_vitis/>`_ (specializes ``ament_hardware_acceleration``)
+     - ``ROS_VITIS``
+     - This CMake variable will evaluate to ``True`` when targeting ROS 2-enabled accelerators that use the Vitis platform for hardware acceleration.
+
+..   * - 
+     - ``ROS_XRT``
+     -  This CMake variable will evaluate to ``True`` if the Xilinx Runtime (XRT) Library  is enabled in the build system. This variable is meant to be used by the maintainers to conditionally compile their ROS 2 packages for acceleration kernels that leverage XRT.
+
+
+Through ``ament_hardware_acceleration`` and technology-specific specializations (like ``ament_vitis``), the ROS 2 build system is automatically enhanced to support producing  acceleration kernel and related artifacts as part of the build process when invoking ``colcon build``. To facilitate the work of maintainers, this additional functionality is configurable through ``mixins`` that can be added to the build step of a ROS 2 workspace, triggering all the hardware acceleration logic only when appropriate (e.g. when ``--mixin kv260`` is appended to the build effort, it'll trigger the build of kernels targeting the KV260 hardware solution). For a reference implementation of these enhacements, refer to `ament_vitis <https://github.com/ros-acceleration/ament_vitis/>`_.
+
+In turn, additional extensions to the existing CMake macros might be proposed which would allow to support more technologies and hardware solutions. For example, ``ament_vitis`` provides a ``vitis_acceleration_kernel`` macro that can be used to generate kernels for the Xilinx Vitis platform. Similarly, ``ament_rocm`` could provide a ``rocm_acceleration_kernel`` macro that can be used to generate kernels for the AMD ROCm platform. This way, maintainers can choose to use the generic ``acceleration_kernel`` macro, or instead use the technology-specific macros to target specific hardware solutions. Also, future extensions to the ``ament_hardware_acceleration`` package could be proposed to support the use of accelerators in binary formats, instead of build them from source code. This would allow to support accelerators which are are not fully integrated into ROS 2 through their corresponding technology libraries (e.g. FPGAs that are not supported by Vitis).
```

Comment:
```suggestion
In turn, additional extensions to the existing CMake macros might be proposed which would allow to support more technologies and hardware solutions. For example, ``ament_vitis`` provides a ``vitis_acceleration_kernel`` macro that can be used to generate kernels for the Xilinx Vitis platform. Similarly, ``ament_rocm`` could provide a ``rocm_acceleration_kernel`` macro that can be used to generate kernels for the AMD ROCm platform. This way, maintainers can choose to use the generic ``acceleration_kernel`` macro, or instead use the technology-specific macros to target specific hardware solutions. Also, future extensions to the ``ament_hardware_acceleration`` package could be proposed to support the use of accelerators in binary format instead of source builds. This would allow to support accelerators which are are not fully integrated into ROS 2 through their corresponding technology libraries (e.g. FPGAs that are not supported by Vitis).
```